### PR TITLE
Replace FULLYPOLARIZED by COLLINEAR[ndown=0]

### PR DESCRIPTION
**Pull request by lukas-weber**
Previous support for fully polarized calculations was not great.

The `FULLYPOLARIZED` walker_type, which was meant to catch the ndown=0 case required
lots of special casing in the code, of which only a small fraction was actually implemented.

This PR removes `FULLYPOLARIZED` and then makes sure that COLLINEAR just works with `ndown == 0`.
The biggest (albeit imperfect) tool to make sure this is the case is a new test `test_polarized_consistency`, which
compares `COLLINEAR[ndown=0]` calculations to `NONCOLLINEAR` calculations with the same orbitals,
which are pretty sure to just work. This case covers all unit test files, which is not perfect but should cover most cases.

Running SAFIRE with a fullypolarized wavefunction file will abort with a warning to regenerate the example as a collinear trial. Trying to generate a fullypolarized wavefunction with afqmctools will print a similar warning.

Many of these changes, especially the `[revisit!]` commit can be made simplified by hardening nda against zero-extent corner cases.

### CODE REVIEW

File: tests/test_polarized_consistency.cpp
Comment (bkesk) [inline tests/test_polarized_consistency.cpp:1]: Good test!

File: utils/afqmctools/utils/slater_types.py [hunk 1]
```diff
@@ -61,7 +62,11 @@ def _slater_enum_map(type):
     elif type in (3,'noncollinear','ghf',SpinSymm.NONCOLLINEAR):
         return _SlaterType.NONCOLLINEAR
     elif type in (4,'fully_polarized','fullypolarized','fp'):
```
Comment (bkesk) [inline utils/afqmctools/utils/slater_types.py]: In practice, I don't think anyone is using fully polarized walkers. Perhaps we should go ahead and completely remove it now and treat is as an Invalid Slater type.

File: utils/afqmctools/utils/types.py [hunk 2]
```diff
@@ -38,7 +38,11 @@ def get_spin_symm_enum(spin_symm):
         return SpinSymm.COLLINEAR
     elif spin_symm in ('ghf','noncollinear',"nc",SpinSymm.NONCOLLINEAR,3):
         return SpinSymm.NONCOLLINEAR
-    elif spin_symm in ('fullypolarized','fully polarized','fp',SpinSymm.FULLYPOLARIZED,4):
-        return SpinSymm.FULLYPOLARIZED
+    elif spin_symm in ('fullypolarized','fully polarized','fully_polarized','fp',4):
```
Comment (bkesk) [inline utils/afqmctools/utils/types.py]: Same comment as before. Just flagging this as another spot to remove this.

**Review by bkesk — APPROVED** [review]
Feel free to merge the PR, I would just suggest removing the fully polarized walker warnings on the Python side in favor of treating them as an invalid type first. I don't think anyone is using fully polarized walkers in practice.
